ftdetect/hack.vim

```diff
@@ -0,0 +1,31 @@
+" Vim ftdetect file
+" Language:     Hack (PHP)
+" Maintainer:   Max Wang <mwang@fb.com>
+" URL:          https://github.com/hhvm/vim-hack
+" Last Change:  April 3, 2014
+"
+" Copyright: (c) 2014, Facebook Inc.  All rights reserved.
+"
+" This source code is licensed under the MIT license found in the
+" LICENSE file found at https://github.com/hhvm/vim-hack.
+"
+au BufRead,BufNewFile *.hack,*.hackpartial
+  \ setl filetype=hack |
+  \ setl syntax=hack
+
+au BufRead,BufNewFile *.hhi,*.hck
+  \ setl filetype=hack |
+  \ setl syntax=php
```

Comment:
If it doesn't use the `hack` syntax, it doesn't belong here? Proper filetype support should go upstream (Vim).

Reply:
Let's just remove the ftdetect.

Reply:
hack is already on the tracking issue.

Reply:
fwiw, this is historical: Hack used to be a superset of PHP, so in hhvm/vim-hack, we set filetype=hack to mark what tooling should be used, but we dealt with the Hack syntax by extending the PHP syntax.

It's not a good way to do things nowadays, and I'm happy to see the syntax=php isn't in the final version of this PR :)

That said: `.hhi` should be treated as a hack file, and probably should have the `hack` syntax. They're not used often - they're effectively our equivalent of header files, and used to give static analysis information about the builtin functions that are implemented in C++.